### PR TITLE
Add Coast Guard Court of Criminal Appeals

### PR DESCRIPTION
**Pull request by flooie**
Adds support for Coast Guard Court of Criminal Appeals.

Based on the work started @EvandoBlanco at https://github.com/freelawproject/juriscraper/pull/376

### CODE REVIEW

File: tests/examples/opinions/united_states/cgcca_example.compare.json [hunk 1]
```diff
@@ -0,0 +1,182 @@
+[
+  {
+    "case_dates": "2021-12-10",
+    "case_names": "United States v. Thomas",
+    "download_urls": "https://media.defense.gov/2021/Dec/13/2002907787/-1/-1/0/UNITED STATES V THOMAS OPINION (MERITS).PDF",
+    "precedential_statuses": "Unpublished",
+    "blocked_statuses": false,
+    "date_filed_is_approximate": false,
+    "docket_numbers": "Docket No. 1476",
```
Comment (mlissner) [inline tests/examples/opinions/united_states/cgcca_example.compare.json:9]: We probably don't want it to say "Docket No." right? Better to just say the number or just say "No. XXX"?

File: tests/examples/opinions/united_states/cgcca_example.compare.json [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    "neutral_citations": "",
+    "west_citations": "",
+    "case_name_shorts": "Thomas"
+  },
+  {
+    "case_dates": "2021-10-15",
+    "case_names": "Order in Re Vargas Petition (Per Curiam",
```
Comment (mlissner) [inline tests/examples/opinions/united_states/cgcca_example.compare.json:16]: This and the next one have hanging parens?

**Review by mlissner — CHANGES_REQUESTED** [review]
Couple things...

File: tests/examples/opinions/united_states/cgcca_example.compare.json (hunk 2 above)
Comment (flooie) [reply tests/examples/opinions/united_states/cgcca_example.compare.json:16]: so much for my streak of no comments

File: tests/examples/opinions/united_states/cgcca_example.compare.json (hunk 1 above)
Comment (flooie) [reply tests/examples/opinions/united_states/cgcca_example.compare.json:9]: sure that seems right

File: tests/examples/opinions/united_states/cgcca_example.compare.json (hunk 2 above)
Comment (flooie) [reply tests/examples/opinions/united_states/cgcca_example.compare.json:16]: √.

**Comment by flooie** [comment]
Thanks @mlissner please let me know.

File: tests/examples/opinions/united_states/cgcca_example.compare.json [hunk 3]
```diff
@@ -66,43 +66,43 @@
...
-    "docket_numbers": "Docket No. 1466",
+    "docket_numbers": "1466",
     "neutral_citations": "79 M.J. 825",
     "west_citations": "",
     "case_name_shorts": ""
   },
   {
     "case_dates": "2020-01-15",
-    "case_names": "United States v. Steen (Unpublished",
+    "case_names": "United States v. Steen (Unpublished)",
```
Comment (mlissner) [inline tests/examples/opinions/united_states/cgcca_example.compare.json:100]: One more corner case, I think.

File: juriscraper/opinions/united_states/federal_special/cgcca.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        for item in self.html.xpath(path):
+            url = item.xpath(".//td/a/@href")[0]
+            first_cell = item.xpath(".//td/a/text()")[0]
+            docket = re.sub(
+                r"Docket Nos?\.", "", item.xpath(".//td[2]/text()")[0]
+            )
+            date_string = item.xpath(".//td[3]/text()")[0]
+            m = re.search(self.citation_regex, first_cell)
+            if m:
+                mj = m.group("MJ") if m.group("MJ") else ""
```
Comment (arderyp) [inline juriscraper/opinions/united_states/federal_special/cgcca.py:44]: I think python will allow short hand ternary: `mj = m.group("MJ") or ""`
Same for line below.

File: juriscraper/opinions/united_states/federal_special/cgcca.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                    "neutral_citation": mj.strip("()"),
+                }
+            )
+
+    def _get_west_citations(self) -> List[str]:
+        """Get West Citations
+
+        :return: List of citations if any
+        """
+        return [c["west_state_citation"].strip("()") for c in self.cases]
```
Comment (arderyp) [inline juriscraper/opinions/united_states/federal_special/cgcca.py:74]: aren't you already doing this strip in line 64?

**Review by arderyp — COMMENTED** [review]
I made some nit-picky suggestions, but looks good in general.

On a separate note, I LOVE that juriscraper is type hinted now :balloon:

File: juriscraper/opinions/united_states/federal_special/cgcca.py (hunk 5 above)
Comment (flooie) [reply juriscraper/opinions/united_states/federal_special/cgcca.py]: yeah you were right - but this needs a second look.  thanks for the comment.

File: tests/examples/opinions/united_states/cgcca_example.compare.json (hunk 3 above)
Comment (flooie) [reply tests/examples/opinions/united_states/cgcca_example.compare.json]: yep.  things should now be improved.
